### ISSUE TITLE
test_improper failing on i686 arch

### ISSUE THREAD

**Issue by rathann** [MDAnalysis/mdanalysis#271]
After getting MDAnalysis and *Tests built and installed, running the testsuite results in this:

/usr/bin/python2 setup.py nosetests -w MDAnalysisTests
[...]
======================================================================
FAIL: test_improper (MDAnalysisTests.test_topology.TestTopologyObjects)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILDROOT/python-MDAnalysis-0.9.2-3.fc23.i386/usr/lib/python2.7/site-packages/MDAnalysisTests/test_topology.py", line 394, in test_improper
    assert_almost_equal(imp.improper(), -3.8370631, 4)
  File "/usr/lib/python2.7/site-packages/numpy/testing/utils.py", line 490, in assert_almost_equal
    raise AssertionError(_build_err_msg())
AssertionError: 
Arrays are not almost equal to 4 decimals
 ACTUAL: -3.836961
 DESIRED: -3.8370631
-------------------- >> begin captured logging << --------------------
MDAnalysis.topology.PSF: DEBUG: PSF file /builddir/build/BUILDROOT/python-MDAnalysis-0.9.2-3.fc23.i386/usr/lib/python2.7/site-packages/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /builddir/build/BUILDROOT/python-MDAnalysis-0.9.2-3.fc23.i386/usr/lib/python2.7/site-packages/MDAnalysisTests/data/adk_dims.dcd...
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1422 tests in 378.219s

FAILED (SKIP=17, failures=1)

I'm building/testing MDAnalysis-0.9.2 release on Fedora development branch (F23/rawhide) with the following version of its dependencies installed:
Cython-0.22-1.fc23.i686
numpy-1.9.2-1.fc23.i686
python-setuptools-15.2-1.fc23.noarch
python-devel-2.7.9-11.fc23.i686
netcdf4-python-1.1.6-1.fc23.i686
python-biopython-1.65-1.fc22.i686
python-GridDataFormats-0.2.4-1.fc22.noarch
python-matplotlib-1.4.3-5.fc23.i686
python-networkx-1.9.1-3.fc23.noarch
python-nose-1.3.6-1.fc23.noarch

Note: this doesn't occur on x86_64 arch.

**Comment by rathann** [comment]
Tests also pass on armv7hl: https://kojipkgs.fedoraproject.org//work/tasks/6896/9706896/build.log. The log from failing i686 build is here: https://kojipkgs.fedoraproject.org//work/tasks/9986/9689986/build.log. Note that these are scratch builds and logs will be deleted after a few days.

**Comment by richardjgowers** [comment]
Hi

Thanks for the report.

It looks like it's just failing due to a difference in precision, so I'd say it's safe to ignore it.  

(This is the torsion method on a Torsion object, which uses only numpy internally, so it's likely due to numpy versions and precision?)

**Comment by rathann** [comment]
Isn't the difference too big, though? Can you replicate the same result?
Our numpy is compiled against atlas-3.10.1.

**Comment by richardjgowers** [comment]
Funnily enough, my computer in Uni fails this too, in a different way....

I think what's happened is the test is around an ill conditioned angle (3 degrees in size) and rounding errors are causing an error at the 4th decimal place.

I'll let the guys in the US wake up and have a say though

======================================================================
FAIL: test_improper (MDAnalysisTests.test_topology.TestTopologyObjects)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/richard/code/mdanalysis/testsuite/MDAnalysisTests/test_topology.py", line 397, in test_improper
    assert_almost_equal(imp.improper(), -3.8370631, 4)
  File "/usr/lib/python2.7/dist-packages/numpy/testing/utils.py", line 470, in assert_almost_equal
    raise AssertionError(msg)
AssertionError: 
Arrays are not almost equal to 4 decimals
 ACTUAL: -3.8370123
 DESIRED: -3.8370631

**Comment by orbeckst** [comment]
... the guy in the US has been busy, sorry.

We have a bunch of cases that appear to be limited in the input precision (e.g. coordinates from GRO or XTC files) where we have to reduce the comparison precision to 3 decimals. 

I generally have no problems accepting 3-4 decimal accuracy in the *analysis* of MD simulations because of all the other approximations that we are making — if there was any result that relied on a difference in the 3rd decimal of an angle I would not believe it for a second. (The issue might be a bit different if you were to *simulate* and then you'd worry a bit more about these inaccuracies building up.)

In summary, I suggest we accept the unit tests results as passes and decrease the precision to 3 but add a note to the documentation to say that inaccuracies in the calculations make the results only reliable to 3 decimals.
